onnxruntime/core/providers/coreml/builders/op_builder.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+namespace onnxruntime {
+namespace coreml {
+
+class ModelBuilder;
+
+class IOpBuilder {
+ public:
+  virtual ~IOpBuilder() = default;
+
+  // Add operator related
+ public:
+  // Check if the initializers of this operator need preprocess
+  // which will not be copied
+  virtual void AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const = 0;
+
```

Comment:
This could have a default of doing nothing couldn't it?

Reply:
The default doing nothing is in the BaseOpBuilder, IOpBuilder is the interface only with no impl

---
In reply to: [563462924](https://github.com/microsoft/onnxruntime/pull/6392#discussion_r563462924) [](ancestors = 563462924)

Reply:
It's still an interface even if you have a default impl for AddInitializersToSkip. Either way - not a big deal.